code/ship/ship.h

```diff
@@ -283,6 +284,36 @@ typedef struct cockpit_display_info {
 	int size[2];
 } cockpit_display_info;
 
+// structure to keep track of ship locks
+typedef struct lock_info {
```

Comment:
Love the struct which organizes everything well.  But the multi issue we were afraid of caused by us switching to this struct: the aspect-seekers fired by clients in multiplayer do not home in on their targets because we no longer set.... `Player_ai->current_target_is_locked` .   Without that, the server does not know to set the missiles as locked on target.

This is my suggested work around until we are ready to bump the multi version with all the new multi stuff we are cooking up.  Here's network/multi_obj.cpp, line 283, where the client figures out whether it should tell the server that it's locked on target.
```
if ( Player_ai->current_target_is_locked ){
		out_flags |= OOC_TARGET_LOCKED;
	}
	if ( Player_ai->ai_flags[AI::AI_Flags::Seek_lock] ){	
		out_flags |= OOC_TARGET_SEEK_LOCK;
	}
```
Just before this section, I would add a loop through the players lock_info struct to check for any target that is `locked`  and then sets the `OOC_TARGET_LOCKED` bit if it finds it.

Bonus points if you can do the same for OOC_TARGET_SEEK_LOCK.  I don't know the side effects of not setting this "properly", but it could cause weird AI side effects.

A full fix, as I mentioned before, would be creating a new packet type for multilock info, but that has to wait.

Reply:
As suggested on Discord, current proposal is to merge multilock so current master can better serve as a baseline for the imminent multiplayer version bump. Any objections?